### PR TITLE
Integrate Support for GeoJSON

### LINKED ISSUE #706: Integrate GeoJSON to MapML Elements
- [x] Add the generic mapml2geojson and geojson2mapml functions to M.Util
- [x] Provide convenient versions of them on mapml-viewer/web-map and layer-
    - [x] layer.mapml2geojson(options) - returns layer in geojson format
    - [x] map.geojson2mapml(json, options) - adds json as a layer- in the map, also returns the layer-
- [x] Add the ability to support ~~copy~~/paste layers using GeoJSON
    - Copying a layer from layer control, should copy mapml as _text/plain_ and ~~GeoJSON as _application/geo+json_~~
        - copying as _application/geo+json_ can not be done through the clipboard API. [Reference](https://developer.chrome.com/blog/web-custom-formats-for-the-async-clipboard-api/) 
        - Could possibly include an option to copy as geojson per https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/340
- [x] Update tests
- [x] Update web-map-doc
- [ ] Update experiments

### PR DESCRIPTION
**Pull request by AliyanH**
- [x] Add the generic mapml2geojson and geojson2mapml functions to M.Util https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/c63eb234ce080ea6d2fdcbb99f24fb1b79ee0ccc
- [x] Provide convenient versions of them on mapml-viewer/web-map and layer- https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/076b184ab9395e16cdaee6dbd1b009afa6695340
    - [x] layer.mapml2geojson(options) - returns layer in geojson format
    - [x] map.geojson2mapml(json, options) - adds json as a layer- in the map, also returns the layer-
    - [x] Update Tests
- [x] Add the ability to support ~~copy~~/paste layers using GeoJSON through the context menu https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/295224f76629f03b258235051b231a876ab49017
    - Copying a layer from layer control, should copy mapml as _text/plain_ and ~~GeoJSON as _application/geo+json_~~
        - copying as _application/geo+json_ can not be done through the clipboard API. [Reference](https://developer.chrome.com/blog/web-custom-formats-for-the-async-clipboard-api/) 
        - Could possibly include an option to copy as geojson per https://github.com/Maps4HTML/Web-Map-Custom-Element/issues/340
    - [x] Add tests
- [x]  Add the ability to paste layers using Ctrl+V https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/124fccd246bd74f6cab0fe4e1b76e08d280dc4e1
- [x] Generate and add default extent meta, when json.bbox not present https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/02075ec72c5692d44631bcd02ce966f9162324d1
- [x] Add the ability to drag and drop layer- or geojson strings to the viewer https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/104204f42919e420d3183e7b21b9fe0018a6485a
- [X] Add localized default layer name https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/0362e657e51e87b65d4fc79640ae3d7b05386b8f
- [x] Add more tests cb17e3e282f040841529fc4656b7abddb228f00d
   - [x] Duplicate Tests for mapml-viewer and web-map
   - [x] For dragDrop Test, change mapml link to use local link instead of geogratis
   - [x] More tests for Ctrl+V pasting
- [x] Paste mapml links through context menu and Ctrl+V cb17e3e282f040841529fc4656b7abddb228f00d
- [ ] Merge the following PR's after this PR is merged:
   - [ ] https://github.com/Maps4HTML/experiments/pull/107
   - [ ] https://github.com/Maps4HTML/web-map-doc/pull/95
   - [ ] https://github.com/Maps4HTML/mapml-extension/pull/47

Closes #706

### CODE REVIEW

**Comment by prushforth** [comment]
Would it be possible to calculate the layer extent as we process the feature geometries?  i.e., since we have to process each and every coordinate in the json, we could keep a running min / max and set that via a \<map-meta> at the end of the process?

**Comment by AliyanH** [comment]
> Would it be possible to calculate the layer extent as we process the feature geometries?

Yes, I believe this can be done.

> and set that via a `<map-meta>` at the end of the process?

It should only calculate the layer's extent when json.bbox is not present (I already map the json.bbox to a map-meta) I presume?

**Comment by prushforth** [comment]
> It should only calculate the layer's extent when json.bbox is not present (I already map the json.bbox to a map-meta) I presume?

I think that's correct.

**Comment by prushforth** [comment]
This is working like a charm, thank you!  Let's review together tomorrow.

File: src/web-map.js [hunk 1]
```diff
@@ -409,6 +423,24 @@ export class WebMap extends HTMLMapElement {
               {target: this}}));
```
Comment (AliyanH) [inline src/web-map.js:423]: @prushforth A lot of the other code in web-map.js is specific to mapml-viewer and most likely does not work on web-map. We will need to go over this in the future. Ex:
`document.activeElement.nodeName === "MAPML-VIEWER"`

Comment (prushforth) [reply src/web-map.js:413]: Whoops! definitely, put it on the backlog...

File: src/mapml-viewer.js [hunk 2]
```diff
@@ -368,6 +372,41 @@ export class MapViewer extends HTMLElement {
...
       }
     });
+    // pasting layer- and geojson using Ctrl+V 
+    this.parentElement.addEventListener('keydown', function (e) {
+      if(e.keyCode === 86 && e.ctrlKey && document.activeElement.nodeName === "MAPML-VIEWER"){
+        navigator.clipboard
+          .readText()
+          .then(
+            (layer) => {
+              try {
```
Comment (AliyanH) [inline src/mapml-viewer.js:382]: This code is replicated a few times, here, for dragDrop, and contextmenu paste. I will create a general function for this instead.

Comment (AliyanH) [reply src/mapml-viewer.js]: Fixed [here](https://github.com/Maps4HTML/Web-Map-Custom-Element/pull/707/commits/5b2444d5fe69c3ff30adb7a9dc8a4d04026a4c15).

**Pull request merged**